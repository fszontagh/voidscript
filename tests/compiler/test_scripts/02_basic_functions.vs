// Test basic function definitions and calls
function greet(string $name) {
    printnl("Hello, ", $name, "!");
}

function add(int $a, int $b) int {
    return $a + $b;
}

function multiply(int $x, int $y) int {
    int $result = $x * $y;
    return $result;
}

// Function calls
greet("VoidScript");
int $sum = add(5, 3);
printnl("Sum: ", $sum);

int $product = multiply(4, 7);
printnl("Product: ", $product);

// Nested function calls
int $nested = add(multiply(2, 3), add(1, 2));
printnl("Nested result: ", $nested);