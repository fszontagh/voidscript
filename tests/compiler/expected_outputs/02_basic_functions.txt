Hello, VoidScript!
Sum: 8
Product: 28
Nested result: 9
